skrifa/src/outline/autohint/latin/blues.rs

```diff
@@ -436,6 +435,153 @@ fn compute_default_blues(font: &FontRef, coords: &[F2Dot14], style: &StyleClass)
+        if is_horizontal {
+            continue;
+        }
+        let is_right = blue_flags & blue_flags::CJK_RIGHT != 0;
+        let is_top = blue_flags & blue_flags::TOP != 0;
+        let blues = &mut blues[!is_horizontal as usize];
+        if blues.len() >= MAX_BLUES {
+            continue;
+        }
+        let mut n_flats = 0;
+        let mut n_fills = 0;
+        let mut is_fill = true;
+        for ch in *blue_chars {
+            if *ch == '|' {
+                is_fill = false;
+                continue;
+            }
+            // TODO: do some shaping
+            let Some(gid) = charmap.map(*ch) else {
+                continue;
+            };
+            if gid.to_u32() == 0 {
+                continue;
+            }
+            let Some(glyph) = glyphs.get(gid) else {
+                continue;
+            };
+            outline_buf.clear();
+            if glyph.draw_unscaled(coords, None, &mut outline_buf).is_err() {
+                continue;
+            }
+            let outline = outline_buf.as_ref();
+            // Reject glyphs that don't produce any rendering
+            if outline.points.len() <= 2 {
+                continue;
+            }
+            let mut best_pos: Option<i16> = None;
+            // Find the extreme point depending on whether this is a top, left,
+            // bottom or right blue
+            if is_horizontal {
+                if is_right {
+                    outline.find_last_contour(|point| {
+                        if best_pos.is_none() || Some(point.x) > best_pos {
+                            best_pos = Some(point.x);
+                            true
+                        } else {
+                            false
+                        }
+                    });
+                } else {
+                    outline.find_last_contour(|point| {
+                        if best_pos.is_none() || Some(point.x) < best_pos {
+                            best_pos = Some(point.x);
+                            true
+                        } else {
+                            false
+                        }
+                    });
+                }
+            } else if is_top {
+                outline.find_last_contour(|point| {
+                    if best_pos.is_none() || Some(point.y) > best_pos {
+                        best_pos = Some(point.y);
+                        true
+                    } else {
+                        false
+                    }
+                });
+            } else {
+                outline.find_last_contour(|point| {
+                    if best_pos.is_none() || Some(point.y) < best_pos {
+                        best_pos = Some(point.y);
+                        true
+                    } else {
+                        false
+                    }
+                });
+            }
+            let Some(best_pos) = best_pos else {
```

Comment:
You checked if the outline has points above. find_last_contour walks all the points. Unless I'm missing something this can't be None.

Reply:
(resolving because this goes away if you reduce)